connector.go

```diff
@@ -35,9 +34,12 @@ func encodeConnectionAttributes(textAttributes string) string {
 	connAttrsBuf = appendLengthEncodedString(connAttrsBuf, connAttrPlatformValue)
 	connAttrsBuf = appendLengthEncodedString(connAttrsBuf, connAttrPid)
 	connAttrsBuf = appendLengthEncodedString(connAttrsBuf, strconv.Itoa(os.Getpid()))
+	connAttrsBuf = appendLengthEncodedString(connAttrsBuf, connAttrServerHost)
+	serverHost, _, _ := net.SplitHostPort(cfg.Addr)
+	connAttrsBuf = appendLengthEncodedString(connAttrsBuf, serverHost)
```

Comment:
Is this correct for Unix socket path?

Reply:
If it's using unix socket, `net.SplitHostPort(cfg.Addr)` will return an empty host value, so the `_server_host` attribute value will be empty, which I think is reasonable, considered the definition is:

![image](https://github.com/go-sql-driver/mysql/assets/61085039/cb15298b-f38c-4972-b32f-815aa0546dcd)

![image](https://github.com/go-sql-driver/mysql/assets/61085039/9500bbcd-475c-493c-bea4-c9dea220cea9)


The MariaDB (Connector/C) client also passes an empty `_server_host` if it's using unix socket:
```bash
$ mysql -S /run/mysqld/mysqld.sock -e "SELECT * FROM performance_schema.session_connect_attrs WHERE processlist_id=CONNECTION_ID();"
+----------------+-----------------+------------+------------------+
| PROCESSLIST_ID | ATTR_NAME       | ATTR_VALUE | ORDINAL_POSITION |
+----------------+-----------------+------------+------------------+
|          13218 | _os             | Linux      |                0 |
|          13218 | _client_name    | libmariadb |                1 |
|          13218 | _pid            | 858742     |                2 |
|          13218 | _client_version | 3.1.20     |                3 |
|          13218 | _platform       | x86_64     |                4 |
|          13218 | program_name    | mysql      |                5 |
+----------------+-----------------+------------+------------------+
$ mysql -h localhost -e "SELECT * FROM performance_schema.session_connect_attrs WHERE processlist_id=CONNECTION_ID();"
+----------------+-----------------+------------+------------------+
| PROCESSLIST_ID | ATTR_NAME       | ATTR_VALUE | ORDINAL_POSITION |
+----------------+-----------------+------------+------------------+
|          13219 | _os             | Linux      |                0 |
|          13219 | _client_name    | libmariadb |                1 |
|          13219 | _pid            | 858747     |                2 |
|          13219 | _client_version | 3.1.20     |                3 |
|          13219 | _platform       | x86_64     |                4 |
|          13219 | program_name    | mysql      |                5 |
|          13219 | _server_host    | localhost  |                6 |
+----------------+-----------------+------------+------------------+
$  mysql -h 127.0.0.1 -e "SELECT * FROM performance_schema.session_connect_attrs WHERE processlist_id=CONNECTION_ID();"
+----------------+-----------------+------------+------------------+
| PROCESSLIST_ID | ATTR_NAME       | ATTR_VALUE | ORDINAL_POSITION |
+----------------+-----------------+------------+------------------+
|          13268 | _os             | Linux      |                0 |
|          13268 | _client_name    | libmariadb |                1 |
|          13268 | _pid            | 859150     |                2 |
|          13268 | _client_version | 3.1.20     |                3 |
|          13268 | _platform       | x86_64     |                4 |
|          13268 | program_name    | mysql      |                5 |
|          13268 | _server_host    | 127.0.0.1  |                6 |
+----------------+-----------------+------------+------------------+
$ mysql --version
mysql  Ver 15.1 Distrib 10.3.38-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2
```

Reply:
> The MariaDB (Connector/C) client also passes an empty _server_host if it's using unix socket:

How did you distinguish mariadb sends empty `_server_host` instead of not sending `_server_host`?

Reply:
Sorry for the uncertainty, you were right, Connector/C doesn't send the `_server_host` attr if unix socket is used. I guess I assumed it does because some other connectors do:

https://github.com/mariadb-corporation/mariadb-connector-j/blob/master/src/main/java/org/mariadb/jdbc/message/client/HandshakeResponse.java#L99-L100

https://github.com/mariadb-corporation/mariadb-connector-r2dbc/blob/4f9385fd9cabcab4557c1f228cfdafa4a3a27ce2/src/main/java/org/mariadb/r2dbc/message/client/HandshakeResponse.java#L173-L174

However it seems the Connector/Nodejs doesn't send the `_server_host` in this case either

https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/master/lib/cmd/handshake/auth/handshake.js#L139-L142

I will add a check before adding the `_server_host` attribute.